app/src/main/java/com/example/sunmusic/action/service/MusicService.kt

```diff
@@ -1,23 +1,46 @@
 package com.example.sunmusic.action.service
 
 import android.app.Service
+import android.content.BroadcastReceiver
+import android.content.Context
 import android.content.Intent
+import android.content.IntentFilter
 import android.media.MediaPlayer
 import android.net.Uri
 import android.os.*
-import android.util.Log
-import android.widget.Toast
 import com.example.sunmusic.action.broadcast.BroadcastManager
 import com.example.sunmusic.action.notification.MyNotificationManager
 import com.example.sunmusic.data.model.Song
-import com.example.sunmusic.utils.MusicAction
-import com.example.sunmusic.utils.SONG_EXTRA
+import com.example.sunmusic.utils.*
 
 class MusicService : Service(), MusicCallback {
 
+    override var mPosition: Int = 0
     private var mMediaPlayer: MediaPlayer? = null
-    private var mSong: Song? = null
+    private var mList: ArrayList<Song>? = null
     private val binder = MyBinder()
+    private var myNotificationReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            when (intent.getStringExtra(ACTION_EXTRA)) {
+                MusicAction.PAUSE.name -> {
+                    pauseSong()
+                }
```

Comment:
1 dòng thì viết thế này 
```suggestion
                MusicAction.PAUSE.name -> pauseSong()
```